File: src/main/java/com/uwetrottmann/trakt5/entities/EpisodeIds.java
/*
 * Copyright 2024 Uwe Trottmann
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.uwetrottmann.trakt5.entities;

import javax.annotation.Nonnull;

public class EpisodeIds extends BaseIds {

    public Integer tvdb;
    /**
     * @deprecated This appears to be no longer supported.
     */
    @Deprecated
    public Integer tvrage;

    @Nonnull
    public static EpisodeIds trakt(int trakt) {
        EpisodeIds ids = new EpisodeIds();
        ids.trakt = trakt;
        return ids;
    }

    @Nonnull
    public static EpisodeIds imdb(String imdb) {
        EpisodeIds ids = new EpisodeIds();
        ids.imdb = imdb;
        return ids;
    }

    @Nonnull
    public static EpisodeIds tmdb(int tmdb) {
        EpisodeIds ids = new EpisodeIds();
        ids.tmdb = tmdb;
        return ids;
    }

    @Nonnull
    public static EpisodeIds tvdb(int tvdb) {
        EpisodeIds ids = new EpisodeIds();
        ids.tvdb = tvdb;
        return ids;
    }

    /**
     * @deprecated This appears to be no longer supported.
     */
    @Deprecated
    @Nonnull
    public static EpisodeIds tvrage(int tvrage) {
        EpisodeIds ids = new EpisodeIds();
        ids.tvrage = tvrage;
        return ids;
    }

}

File: src/main/kotlin/com/redissi/trakt/entities/EpisodeIds.kt
package com.redissi.trakt.entities

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
class EpisodeIds(
    override val trakt: Int? = null,
    override val imdb: String? = null,
    override val tmdb: Int? = null,
    val tvdb: Int? = null,
    val tvrage: Int? = null
) : BaseIds() {

    companion object {
        fun trakt(trakt: Int): EpisodeIds {
            return EpisodeIds(trakt = trakt)
        }

        fun imdb(imdb: String?): EpisodeIds {
            return EpisodeIds(imdb = imdb)
        }

        fun tmdb(tmdb: Int): EpisodeIds {
            return EpisodeIds(tmdb = tmdb)
        }

        fun tvdb(tvdb: Int): EpisodeIds {
            return EpisodeIds(tvdb = tvdb)
        }

        fun tvrage(tvrage: Int): EpisodeIds {
            return EpisodeIds(tvrage = tvrage)
        }
    }
}